examples/isolated-functions/isolated_functions.bal

```diff
@@ -0,0 +1,47 @@
+import ballerina/io;
+
+const BALLERINA = "Ballerina";
+
+final string[] & readonly prefixes = ["hello", "greetings", "hello world"];
+
+// An `isolated` function that adds to an array a specific number of
+// greeting strings.
+// The `greetings` parameter is the array to which greetings have to
+// be added. The `count` parameter indicates the number of greetings
+// to add.
+isolated function addGreetings(string[] greetings, int count) {
+    // Call another `isolated` function to build a specific number
+    // of greetings.
+    string[] builtGreetings = getGreetings(count);
+
+    // Add the greetings returned by the `isolated` function to
+    // the mutable array passed as an argument.
+    greetings.push(...builtGreetings);
+}
+
+// An `isolated` function that accesses global immutable state and
```

Comment:
```suggestion
// An `isolated` function, which accesses the global, immutable state and
```

Reply:
State in this context is not singular. Applicable to the other suggestions too.

Reply:
+1